### PR TITLE
shared: fix onEvent typings & add getByScriptID to client Entity

### PR DESCRIPTION
**Pull request by Yiin**

### CODE REVIEW

File: client/src/helpers/NativeInvoker.h [hunk 1]
```diff
@@ -14,24 +14,40 @@ namespace js
...
+            nextString = (nextString + 1) % 256;
 
             return strVal;
         }
 
         template<typename T>
         T* SavePointer(T val)
         {
+            if (pointersCount >= 32) {
+                pointersCount = 0;
```
Comment (xLuxy) [inline client/src/helpers/NativeInvoker.h:48]: don't do that

File: client/src/helpers/NativeInvoker.h [hunk 2]
```diff
@@ -40,6 +56,10 @@ namespace js
         template<>
         alt::INative::Vector3* SavePointer(alt::INative::Vector3 val)
         {
+            if (pointersCount + 2 >= 32) {
+                pointersCount = 0;
```
Comment (xLuxy) [inline client/src/helpers/NativeInvoker.h:60]: don't do that

File: client/src/helpers/NativeInvoker.h [hunk 3]
```diff
@@ -14,24 +14,40 @@ namespace js
...
         int stringsCount = 0;
         int returnsCount = 1;
 
         char* SaveString(const std::string& str)
         {
+            static int nextString = 0;
+
+            // Free memory of the string that's about to be overwritten
+            if (stringValues[nextString])
+                free(stringValues[nextString]);
```
Comment (xLuxy) [inline client/src/helpers/NativeInvoker.h:29]: You free memory here, but when will you free the actual memory once invocations is done?

**Review by xLuxy — CHANGES_REQUESTED** [review]

**Comment by LeonMrBonnie** [comment]
Just remove all the changes to natives, I will fix that myself

**Comment by Yiin** [comment]
@LeonMrBonnie done

**Comment by LeonMrBonnie** [comment]
i forgor

**Pull request merged**